test/libsolidity/semanticTests/variables/transient_function_type_state_variable.sol

```diff
@@ -0,0 +1,26 @@
+contract C {
+    function () external transient f;
+    function () external f1;
+    function g() external {
+    }
+
+    function testTransient() public returns (bool) {
+        assert(f != this.g);
+        f = this.g;
+
+        return f == this.g;
+    }
+
+    function testStorage() public returns (bool) {
+        assert(f1 != this.g);
+        f1 = this.g;
+
+        return f1 == this.g;
+    }
+}
+// ====
+// EVMVersion: >=cancun
+// compileViaYul: false
+// ----
+// testTransient() -> false
```

Comment:
I checked with a small example and the asm produced by the compiler is the same in both cases, transient and storage, with the only difference that `tstore` and `tload` are used instead of `sstore` and `sload`.
Maybe this is related to evmone?

Reply:
This was actually a missing change in the PR. Fixed.